### ISSUE TITLE
Screenshots can not taken when using az machinery

### ISSUE THREAD

**Issue by thezolin** [kevoreilly/CAPEv2#1172]
## About accounts on [capesandbox.com](https://capesandbox.com/)
* Issues isn't the way to ask for account acctivation. Ping capesandbox in [Twitter](https://twitter.com/capesandbox) with your username

## This is opensource and you getting __free__ support so be friendly!
* Free support from doomedraven ended, no whiskey no support. For something he updated the documentation :)

# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I did read the README!
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed
- [X] I'm reporting the issue to the correct repository (for multi-repository projects)
- [X ] I'm have read all configs with all optional parts

# Expected Behavior

Please describe the behavior you are expecting. __If your samples(x64) stuck in pending ensure that you set tags=x64 in hypervisor conf for x64 vms__

Take screenshots during analysis phase. 

# Current Behavior

What is the current behavior?
Screenshots cannot be taken

# Failure Information (for bugs)

It's not a failure, but a strange behaviour. 

## Steps to Reproduce

When I use azure as a machine, I had to create a basic task with agent.pyw to be started during vm startup.

However, the analysis does happen, just without screenshots.

If I change the agent to "When I login", the azure only sees the machine if I login with that account, because the agent only runs at login.

Following the instructions in the documentation, CAPE does not identify the machine ready for analysis, since the agent is not running on that VMSS. If I log on, the script runs and the screenshot is taken.

If I use the other method (on startup), the analysis happens without the screenshot.

Is there any suggestion to reconcile the two ways? When used in kvm this behavior does not happen.

## Context

CAPE deployed under azure

## Failure Logs

Please include any relevant log snippets or files here.

**Comment by kevoreilly** [comment]
A difference with kvm is that there snapshots of a running state are used as opposed to booting every job.

Perhaps the 'on startup' method launches the agent without elevation?

I may have to defer to @cccs-kevin as I lack any real azure knowledge.

**Comment by thezolin** [comment]
![azure_agent_s1](https://user-images.githubusercontent.com/36763110/193446238-da5decd4-7ac6-4660-93ee-43637959e5da.png)

On startup method I ensure "run with highest privileges" is checked. I have tried to combine two tasks, but no success yet. 

Thanks for your help, @kevoreilly

**Comment by cccs-kevin** [comment]
Hey @thezolin, it sounds like you have two problems:

1. The CAPE agents on the VMSS VMs are not starting correctly once the VMs are booted, depending on your setup of the agent
2. Screenshots are not taken on the VMSS VM

For the first issue, Azure "snapshots" do not maintain the state of the VM (running, not running) and are instead copies of the file system at a certain point in time. Therefore for each new task, a VM is essentially booted from a powered off state using this snapshot, so there is a slight time delay for the OS to spin up and the agent to start listening.

If you are using the startup method via the Task Scheduler, you definitely want to select "Run whether user is logged on or not" since that is the only way that the agent will start when the VM is booted up. Selecting "Run with highest privileges" can't hurt either.

Alternatively if that doesn't work (which it seems like it isn't), you can try placing the agent file in the directory specified [here](https://capev2.readthedocs.io/en/latest/installation/guest/agent.html#prior-to-windows-10), which works for even some Windows 10 VMs.

Regarding the second issue, "the analysis happens without the screenshot", I think the answer would lie in the `analysis.log` generated for the analysis which you can find on the CAPE host at /opt/CAPEv2/storage/analyses/<task-id>/analysis.log. This is a log of everything that is happening on the VMSS VM, and should provide some insight into why the screenshots are not being taken.

**Comment by thezolin** [comment]
Hi @kevoreilly!
I put the agent in the specific directory, but no success. I believe is because of windows version (Windows 10 Enterprise 21H2)
I had checked the analysis.log file and the error message have no clue: 
2022-10-05 06:35:34,256 [lib.api.screenshot] DEBUG: Importing 'PIL.ImageChops'
2022-10-05 06:35:34,412 [lib.api.screenshot] DEBUG: Importing 'PIL.ImageGrab'
2022-10-05 06:35:34,412 [lib.api.screenshot] DEBUG: Importing 'PIL.ImageDraw'
...
2022-10-05 06:35:34,459 [root] DEBUG: Initialized auxiliary module "Screenshots"
2022-10-05 06:35:34,459 [root] DEBUG: Trying to start auxiliary module "Screenshots"...
2022-10-05 06:35:34,459 [root] DEBUG: Started auxiliary module "Screenshots"
2022-10-05 06:35:34,459 [root] DEBUG: Started auxiliary module Screenshots
...
2022-10-05 06:35:36,490 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2022-10-05 06:35:36,506 [modules.auxiliary.evtx] DEBUG: Wiping Microsoft-Windows-Sysmon/Operational
2022-10-05 06:35:37,506 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2022-10-05 06:35:38,515 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
2022-10-05 06:35:39,530 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed

If I run agent using "When I logon" trigger, and login on VM, works fine. 
If I run agent on startup, login or not, no screenshot taken. :(

Thanks again for your help.

**Comment by cccs-kevin** [comment]
Hmmm okay, this error is occurring [here ](https://github.com/kevoreilly/CAPEv2/blob/master/analyzer/windows/modules/auxiliary/screenshots.py#L53) in the Screenshots module, so maybe try tweaking that log to get more information?

Maybe try enabling `netplwiz` to auto-login? https://www.howtogeek.com/443119/how-to-enable-or-disable-secure-sign-in-for-windows-10/

**Comment by thezolin** [comment]
Hi @cccs-kevin, 
After this argument has been added, I got this error message:
2022-10-07 05:51:43,761 [modules.auxiliary.screenshots] ERROR: Cannot take screenshot: screen grab failed
Traceback (most recent call last):
  File "C:\tmpyotht4tt\modules\auxiliary\screenshots.py", line 51, in run
    img_current = Screenshot().take()
  File "C:\tmpyotht4tt\lib\api\screenshot.py", line 92, in take
    return ImageGrab.grab() if HAVE_PIL else None
  File "C:\Program Files (x86)\Python310-32\lib\site-packages\PIL\ImageGrab.py", line 46, in grab
    offset, size, data = Image.core.grabscreen_win32(
OSError: screen grab failed

I have tried the auto-login functionality, but still no success. 
![Screenshot_23](https://user-images.githubusercontent.com/36763110/194479028-f7dbe271-438e-4fbb-a7c8-5e4c931a3681.png)

**Comment by cccs-kevin** [comment]
Hey @thezolin thanks for adding that exception info to the log. Unfortunately it doesn't help us debug the screenshot issue. BUT by looking at your CAPE logs, it is definitely confirmed that for whatever reason, the agent is not accessible by CAPE, which we've already discussed. 

If you follow the docs [here](https://capev2.readthedocs.io/en/latest/installation/guest/agent.html#installing-the-agent), this part may be useful to you:

```
Don’t forget to test the agent before saving the snapshot. You can do it ... by executing: curl VM_IP:8000. You should see an output similar to the following:
```
![image](https://user-images.githubusercontent.com/59843993/194545639-99a2831d-b510-42e6-aad2-59f5db3be8ad.png)

Basically I think the issue lies in your "golden image" VM, and I would start the "golden image" VM from a powered-off state, wait 30 seconds, then test this curl command from the host, and if the host does not get the expected response, then something if not working correctly on the "golden image" VM. This way you can isolate the problem to how your "golden image" VM is working prior to taking a snapshot and prior to running CAPE, which should make the problem easier.

Good luck!
